packages/openmrs/src/Utils.js

```diff
@@ -1,44 +1,36 @@
 import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  makeBasicAuthHeader,
+} from '@openfn/language-common/util';
 
-export const Log = {
-  success: message => console.log(`✓ Success at ${new Date()}:\n∟ ${message}`),
-  warn: message => console.log(`⚠ Warning at ${new Date()}:\n∟ ${message}`),
-  error: message => console.log(`✗ Error at ${new Date()}:\n∟ ${message}`),
-  info: message => console.log(`ℹ Info at ${new Date()}:\n∟ ${message}`),
+export const prepareNextState = (state, response, callback) => {
+  const nextState = {
+    ...composeNextState(state, response.body),
+    response,
+  };
+
+  return callback(nextState);
 };
 
-export function handleError(error) {
-  if (error.response) {
-    const { method, path } = error.response.req;
-    const { status } = error.response;
-    if (Object.keys(error.response.body).length === 0) {
-      throw new Error(
-        `Server responded with:  \n${JSON.stringify(error.response, null, 2)}`
-      );
-    }
-
-    const errorString = [
-      `Request: ${method} ${path}`,
-      `Got: ${status}`,
-      'Body:',
-      JSON.stringify(error.response.body, null, 2),
-    ].join('\n');
-
-    throw new Error(errorString);
-  } else {
-    throw error;
-  }
-}
+export function request(state, method, path, data, params) {
+  const { instanceUrl, username, password } = state.configuration;
+  const headers = makeBasicAuthHeader(username, password);
 
-export function handleResponse(response, state, callback) {
-  const { body } = response;
-  const nextState = composeNextState(state, { body });
-  if (callback) return callback(nextState);
-  return nextState;
-}
+  const options = {
+    body: data,
+
+    headers: {
+      ...headers,
+      'content-type': 'application/json',
+    },
+
+    query: params,
+
+    parseAs: 'json',
+  };
 
-const isArray = variable => !!variable && variable.constructor === Array;
+  const url = `${instanceUrl}${path}`;
 
-export function nestArray(data, key) {
-  return isArray(data) ? { [key]: data } : data;
+  return commonRequest(method, url, options);
```

Comment:
One more thought here.

If you take a look at the HTTP adaptor, it calls `logResponse` from `common` so that every single http request will log the URL and status. 

Working from something Taylor said, maybe we can copy that code out and `console.debug` with similar output? Logging it to debug de-emphasises it (users will be able to hide debug logs one day).

Reply:
Of course - I'll get back to you a bit later